src/filesystem/impls/filer/BlobUtils.js

```diff
@@ -0,0 +1,142 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true, regexp: true, indent: 4, maxerr: 50 */
+/*global define, Blob, URL */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    // BlobUtils provides an opportunistic cache for BLOB Object URLs
+    // which can be looked-up synchronously.
+
+    var Filer = require("thirdparty/filer/dist/filer");
+    var Path  = Filer.Path;
+
+    // 2-way cache for blob URL to path for looking up either way:
+    // * urls - paths keyed on blobUrls
+    // * blobs - blobUrls keyed on paths
+    var urls  = {};
+    var blobs = {};
+
+    // Get a mime-type for the given extension, defaulting to
+    // application/octent-stream if we don't know what it is.
+    function extensionToMime(ext) {
+        switch(ext) {
+        case ".html":
+        case ".htmls":
+        case ".htm":
+        case ".htx":
+            return "text/html";
+        case ".ico":
+            return "image/x-icon";
+        case ".bmp":
+            return "image/bmp";
+        case ".css":
+            return "text/css";
+        case ".js":
+            return "text/javascript";
+        case ".svg":
+            return "image/svg+xml";
+        case ".png":
+            return "image/png";
+        case ".jpg":
+        case ".jpe":
+        case ".jpeg":
+            return "image/jpeg";
+        case ".gif":
+            return "image/gif";
+        // Some of these media types can be video or audio, prefer video.
+        case ".mp4":
+            return "video/mp4";
+        case ".mpeg":
+            return "video/mpeg";
+        case ".ogg":
+        case ".ogv":
+            return "video/ogg";
+        case ".mov":
+        case ".qt":
+            return "video/quicktime";
+        case ".webm":
+            return "video/webm";
+        case ".avi":
+        case ".divx":
+            return "video/avi";
+        case ".mpa":
+        case ".mp3":
+            return "audio/mpeg";
+        case ".wav":
+            return "audio/vnd.wave";
+        default:
+            return "application/octet-stream";    
+        }
+    }
+
+    // Generate a BLOB URL for the given filename and data and cache it
+    function cache(filename, data) {
```

Comment:
Make this async (add callback) and deal with special cases of html or css files, rewriting before storing url.  Pull nohost rewriter.js stuff in here.